include/grpc/impl/codegen/port_platform.h

```diff
@@ -282,6 +282,45 @@
 #else /* _LP64 */
 #define GPR_ARCH_32 1
 #endif /* _LP64 */
+#elif defined(__sun)
```

Comment:
Without further comment on the merits or need of this PR, this macro is not precise since `__sun` can indicate either Solaris or old SunOS targets.

Reply:
I believe `__sun` is actually the most compatible way to check for both Oracle Solaris and Illumos derivatives (SmartOS, etc.) at the moment. I'm having some trouble finding a definitive source on it. However searching in pkgsrc (yeah I know, not the most scientific approach ;-) for [__sun](https://github.com/NetBSD/pkgsrc/search?q=__sun&unscoped_q=__sun) returns 527 hits and [__SVR4](https://github.com/NetBSD/pkgsrc/search?q=__SVR4&unscoped_q=__SVR4) only 24 hits.

FWIW uname also still returns SunOS:

```
$ uname
SunOS
```